test/helpers/transactionConfirmation.js

```diff
@@ -12,10 +13,17 @@ export async function mineBlocks(service, count) {
   if (typeof count !== 'number') {
     count = web3Service.confirmedBlockCount() + 2;
   }
+
   for (let i = 0; i < count; i++) {
     await wethToken.approveUnlimited(TestAccountProvider.nextAddress());
     log(`block ${web3Service.blockNumber()}`);
   }
+
+  // the websocket implementation has a poller somewhere that only fires once
+  // every 4 seconds
+  if (web3Service.usingWebsockets()) {
+    await promiseWait(4000);
```

Comment:
I don't think that's entirely true. I think it's more to do with the delay between ganache mining a block and firing the newBlock event which does mean testing is significantly slower plus other issues. We could abandon the block subscriptions to avoid this and use the polling method in http to improve efficiency.